src/coreclr/jit/importer.cpp

```diff
@@ -4506,54 +4507,41 @@ GenTree* Compiler::impImportStaticFieldAccess(CORINFO_RESOLVED_TOKEN* pResolvedT
 
         default:
         {
-            // Do we need the address of a static field?
-            //
-            if (access & CORINFO_ACCESS_ADDRESS)
+// TODO-CQ: enable this optimization for 32 bit targets.
+#ifdef TARGET_64BIT
+            if (!isBoxedStatic && (lclTyp == TYP_REF))
```

Comment:
TP nit: does this end up calling `getStaticFieldCurrentClass` for `ldsflda` as well?

(There is not much point to that I presume)

Reply:
Good point, in case of ldsflda it's going to be a raw constant anyway. Fixed